### PR TITLE
init(navigation): bottom tab and native stack

### PR DESCRIPTION
**Pull request by samandarkuchkarov**

### CODE REVIEW

File: Joker/src/components/home-screen.title.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import {RouteProp} from '@react-navigation/native';
+import React from 'react';
+import {RootStackParamList} from '../types';
+import {Text} from 'react-native';
+
+export type HomeScreenTitleProps = {
+  route: RouteProp<RootStackParamList>;
```
Comment (gHashTag) [inline Joker/src/components/home-screen.title.tsx:7]: Do not use navigation props, look in the haqq-wallet useTypedRoute hook

File: Joker/src/components/home-screen.title.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import {RouteProp} from '@react-navigation/native';
```
Comment (gHashTag) [inline Joker/src/components/home-screen.title.tsx:1]: Fix file naming home-screen.title.tsx

File: Joker/src/components/tab-bar.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  ParamListBase,
+  TabNavigationState,
+} from '@react-navigation/native';
+import React, {useCallback} from 'react';
+import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+import {IS_IOS} from '../variables';
+
+type TabbarProps = {
+  navigation: NavigationHelpers<ParamListBase, BottomTabNavigationEventMap>;
```
Comment (gHashTag) [inline Joker/src/components/tab-bar.tsx:14]: What kind of NavigationHelpers??
Use typed navigation hook

File: Joker/src/components/tab-bar.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
+// import {RouteProp} from '@react-navigation/native';
```
Comment (gHashTag) [inline Joker/src/components/tab-bar.tsx:1]: remove unnecessary commented line

File: Joker/src/components/tab-bar.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+import {IS_IOS} from '../variables';
+
+type TabbarProps = {
+  navigation: NavigationHelpers<ParamListBase, BottomTabNavigationEventMap>;
+  state: TabNavigationState<ParamListBase>;
+};
+export function TabBar(props: TabbarProps) {
+  const tabRoutes = props.state.routeNames;
```
Comment (gHashTag) [inline Joker/src/components/tab-bar.tsx:18]: Use destructurization 
```js
const { routeNames, index } = props.state;
```

File: Joker/src/components/tab-bar.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  const insets = useSafeAreaInsets();
+  const onPressTab = useCallback(
+    (route: string) => {
+      props.navigation.navigate(route);
+    },
+    [props.navigation],
+  );
+
+  return (
+    <View style={[styles.wrapper, {paddingBottom: Math.max(insets.bottom)}]}>
```
Comment (gHashTag) [inline Joker/src/components/tab-bar.tsx:29]: This construction doesn't make sense right now
```js
Math.max(insets.bottom)
```
[See this](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Math/max)

File: Joker/src/components/tab-bar.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    width: '100%',
+    height: IS_IOS ? 80 : 60,
+    backgroundColor: '#fff',
+    flexDirection: 'row',
+    justifyContent: 'space-around',
+    alignItems: 'center',
+  },
+  tabBarItem: {
+    marginTop: IS_IOS ? 5 : 0,
+    height: IS_IOS ? 50 : 30,
```
Comment (gHashTag) [inline Joker/src/components/tab-bar.tsx:52]: Why do IS_IOS here, because with safe Area In set s you can adjust the lower margins

**Review by gHashTag — COMMENTED** [review]
Add useTypedRoute hook

**Review by gHashTag — CHANGES_REQUESTED** [review]
Fix comments and move navigation to 'src/app`

File: Joker/src/types.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
+export type RootStackParamList = {
+  home: undefined;
```
Comment (gHashTag) [inline Joker/src/types.ts:2]: Look [typescript guide](https://reactnavigation.org/docs/typescript#type-checking-screens-and-params-in-nested-navigator) in react-navigation for nested navigators

File: Joker/src/components/home-screen.title.tsx (hunk 1 above)
Comment (samandarkuchkarov) [reply Joker/src/components/home-screen.title.tsx]: done

File: Joker/src/components/home-screen.title.tsx (hunk 2 above)
Comment (samandarkuchkarov) [reply Joker/src/components/home-screen.title.tsx]: done

File: Joker/src/components/tab-bar.tsx (hunk 3 above)
Comment (samandarkuchkarov) [reply Joker/src/components/tab-bar.tsx:12]: done

**Review by gHashTag — APPROVED** [review]

**Pull request merged**